clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5241,8 +5241,8 @@ bool CheckVectorElementCallArgs(Sema *S, CallExpr *TheCall) {
 
   // Note: if we get here one of the args is a scalar which
   // requires a VectorSplat on Arg0 or Arg1
-  S->Diag(BuiltinLoc, diag::err_vec_builtin_non_vector_all)
-      << TheCall->getDirectCallee()
+  S->Diag(BuiltinLoc, diag::err_vec_builtin_non_vector)
+      << TheCall->getDirectCallee() << /*all args*/ true
```

Comment:
I appreciate the comment about what the parameter means, there is some inconsistency in how it's labeled. After this point, it's always `/*isMoretThanTwoArgs*/` which I think is better.

Reply:
Originally this bool was determined by `(TheCall->getNumArgs() != 2)` Chris preferred the `all` language. So the comment here is to indicate that we are always going to go with all even when arg count is 2. I didn't want to use the same comment because  a future editor might think there is a bug here.

Reply:
I'm going to change the comment to `/*useAllTerminology*/` for the  hlsl cases.